test/azure.cs

```diff
@@ -81,6 +86,41 @@ public async Task small_file_azure()
             Assert.Equal(suffix, url.Substring(url.Length - suffix.Length));
         }
 
+        [Fact(Skip = "Execute manually")]
+        public async Task archive_azure()
+        {
+            var file = FakeFile(1024);
+            var hash = Md5(file);
+            var store = (AzureStore)Store;
+
+            Assert.Equal("B2EA9F7FCEA831A4A63B213F41A8855B", hash.ToString());
+
+            var r = await store.WriteAsync(file, CancellationToken.None);
+            Assert.Equal("B2EA9F7FCEA831A4A63B213F41A8855B", r.Hash.ToString());
+            Assert.Equal(1024, r.Size);
+
+            var a = store[new Hash("B2EA9F7FCEA831A4A63B213F41A8855B")];
+            var aBlob = await a.GetBlob();
+            await store.ArchiveBlobAsync(a);
+            await aBlob.DeleteAsync();
+            await store.TryUnArchiveBlobAsync(new Hash("B2EA9F7FCEA831A4A63B213F41A8855B"));
+
+            Boolean finished = false;
+            while (!finished)
+            {
+                UnArchiveStatus status = await store.TryUnArchiveBlobAsync(new Hash("B2EA9F7FCEA831A4A63B213F41A8855B"));
+                if (status == UnArchiveStatus.Done)
+                {
+                    var a2 = store[new Hash("B2EA9F7FCEA831A4A63B213F41A8855B")];
+                    Assert.Equal("B2EA9F7FCEA831A4A63B213F41A8855B", a2.Hash.ToString());
```

Comment:
This will always be true, without even looking at the contents of the remote file (`store[hash]` just stores the hash inside the returned blob without checking anything). 

You need to load the blob's contents and compare them with variable `file` above.

Reply:
This mistake should be fixed now.